napari/layers/shapes/_tests/test_shapes.py

```diff
@@ -343,45 +343,54 @@ def test_data_setter_with_text(properties):
     assert len(layer.text.values) == n_new_shapes_2
 
 
-def test_rectangles():
+@pytest.mark.parametrize(
+    "shape",
+    [
+        # single & multiple four corner rectangles
+        (1, 4, 2),
+        (10, 4, 2),
```

Comment:
@DragaDoncila I'm highly confused on this "4 corners". You create random arrays of shape (N ellipses , 4 corner , 2 dimension), but nothing enforce those 4 corners to be rectangles. So should not most of those ellipses creation fail ?

Reply:
see https://github.com/napari/napari/issues/4258#issuecomment-1067847258 as well.